### PR TITLE
Add metadata field for backup info

### PR DESCRIPTION
**Pull request by disrani-px**
Signed-off-by: Dinesh Israni <disrani@portworx.com>

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -291,6 +291,8 @@ type BackupInfo struct {
 	// Timestamp is the timestamp at which the source volume
 	// was backed up to cloud
 	Timestamp time.Time
+	// Metadata associated with the backup
```
Comment (lpabon) [inline api/api.go]: Why is this necessary? Could you provide some context?

Comment (disrani-px) [reply api/api.go]: Backups might have extra metadata associated with it. This is used to return that metadata with other backup information.
